infra/src/main/kotlin/fi/fta/geoviite/infra/split/SplitService.kt

```diff
@@ -236,14 +237,63 @@ class SplitService(
             alignment = alignment,
             targets = collectSplitTargetParams(request.targetTracks, suggestions),
         )
-        val splitTargets = targetTracks.map(::saveTargetTrack)
+
+        val savedSplitTargetLocationTracks = targetTracks.map { targetTrack ->
+            targetTrack.copy(
+                locationTrack = targetTrack.locationTrack.copy(
+                    id = saveTargetTrack(targetTrack)
+                ),
+            )
+        }
+
+        geocodingService.getGeocodingContext(DRAFT, sourceTrack.trackNumberId)?.let { geocodingContext ->
+            val splitTargetTracksWithAlignments = savedSplitTargetLocationTracks.map { splitTargetResult ->
+                splitTargetResult.locationTrack to splitTargetResult.alignment
+            }
+
+            updateUnusedDuplicateReferencesToSplitTargetTracks(
+                geocodingContext,
+                request,
+                splitTargetTracksWithAlignments,
+            )
+        }
```

Comment:
Pitäisikö geokoodauskontekstin luonti olla aina edellytetty splittiä tehdessä? (Nyt tämä voi periaatteessa skipata tuon duplikaattireferenssien päivityksen, jos jostain syystä geokoodauskontekstin luonti epäonnistuu.

Reply:
Kuulostaisi tosi oudolta että me lähdettäisiin splittaamaan raidetta jota ei voi geokoodata. Me splitataan käytännössä aina pelkästään OFFICIAL-raiteita, joten se vaatisi sen, että meillä olisi virallisessa paikannuspohjassa raiteita joilla (tai joiden ratanumeroilla) on asiat niin huonosti, että niitä ei voi geokoodata. Mielestäni on siis ihan ok tehdä oletus että jos/kun ollaan päästy splittauksessa näin pitkälle, niin geokoodaus on aina mahdollista (tai jos se ei ole, niin jossain pitäisi heittää poikkeus)